src/__internal__/character-count/character-count.test.tsx

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import CharacterCount from ".";
+
+test("visible character count is visible", () => {
+  render(<CharacterCount value={10} limit={5} isOverLimit={false} />);
+
+  const visibleCharacterCount = screen.getByTestId("character-count");
+  expect(visibleCharacterCount).toBeVisible();
+});
+
+test("visible character count renders with the 'aria-hidden' attribute", () => {
+  render(<CharacterCount value={5} limit={10} isOverLimit={false} />);
+
+  const visibleCharacterCount = screen.getByTestId("character-count");
+  expect(visibleCharacterCount).toHaveAttribute("aria-hidden", "true");
+});
+
+test("visible character count renders with '{count} character(s) left' text content", () => {
+  render(<CharacterCount value={5} limit={10} isOverLimit={false} />);
+
+  const visibleCharacterCount = screen.getByTestId("character-count");
+  expect(visibleCharacterCount).toHaveTextContent("5 characters left");
+});
+
+test("visible character count renders with '{count} character(s) too many' text content when `isOverLimit` prop is true", () => {
+  render(<CharacterCount value={10} limit={5} isOverLimit />);
+
+  const visuallyHiddenHint = screen.getByTestId("character-count");
+  expect(visuallyHiddenHint).toHaveTextContent("5 characters too many");
+});
+
+test("visually hidden character counter is visually hidden", () => {
```

Comment:
**suggestion** (non-blocking): both this test and the other one for visually hidden styles could perhaps be removed, as Chromatic should surely catch it if we accidentally make it visible?